client/src/components/MusicPlayer/MusicPlayer.tsx

```diff
@@ -29,6 +38,67 @@ function MusicPlayer() {
     getUserInfo();
   }, [dispatch]);
 
+  // token persist
+  useEffect(() => {
+    if (sessToken) {
+      const queryString = window.location.search;
+      const urlParams = new URLSearchParams(queryString);
+      const hasError = urlParams.get('error');
+      const persistToken = localStorage.getItem('persist:token');
+      if (persistToken) {
+        const persistParse = JSON.parse(persistToken);
+
+        if (
+          hasError ||
+          hasTokenExpired() ||
+          persistParse.sessToken === 'undefined'
+        ) {
+          getRefreshToken();
+        }
+      }
+    }
+
+    // helper function to check if token expired
+    function hasTokenExpired() {
+      if (!sessToken || !timeStamp) {
+        return false;
+      }
+
+      const millisecondsElpased = Date.now() - Number(timeStamp);
+      return millisecondsElpased / 1000 > Number(tokenExpires);
+    }
+
+    // logout helper function
+    function logout() {
+      window.localStorage.removeItem('persist:token');
+      window.location.assign('/'); // not sure if this works, but trying to jus redirect to origin which should be the login page once localStorage clears
```

Comment:
the origin should be the App.tsx, which should point to the Login component since upon localstorage clear, there shouldn't be a token. still trying to figure out the mechanics on that through redux-persist though.

Reply:
turns out the window.location.assign('/') isn't needed once clearing token on server side.